File: python/quicksort/quicksort.py
def partition(li, start, end):
    pivot = li[start]
    left = start + 1
    right = end
    done = False

    while not done:
        while left <= right and li[left] <= pivot:
            left += 1
        while right >= left and li[right] >= pivot:
            right -= 1
        if right < left:
            done = True
        else:
            li[left], li[right] = li[right], li[left]
    li[start], li[right] = li[right], li[start]
    return right


def quicksort(li, start=0, end=None):
    if not end:
        end = len(li) - 1
    if start < end:
        pivot = partition(li, start, end)
        # Recursively perform same operation on first and last halves of list
        quicksort(li, start, pivot - 1)
        quicksort(li, pivot + 1, end)
    return li

File: algorithms/quick_sort/php/quick_sort.php
<?php
 
    function quicksort($array)
    {
        if (count($array) == 0)
            return array();
 
        $pivot = $array[0];
        $left = $right = array();
 
        for ($i = 1; $i < count($array); $i++) {
            if ($array[$i] <$pivot)
                $left[] = $array[$i];
            else
                $right[] = $array[$i];
        }
 
        return array_merge(quicksort($left), array($pivot), quicksort($right));
    }

    $list = array(9,2,1,28,13,17,13,69);
 
    $quick_sorted_numbs = quicksort($list);
 
    die(json_encode($quick_sorted_numbs));